plugins/reporters/fossid/src/main/resources/templates/asciidoc/fossid_snippet.ftl

```diff
@@ -87,7 +87,7 @@ ${sourceLocation.startLine}-${sourceLocation.endLine}
 [#list snippetFinding.snippets as snippet ]
 [#assign matchType = snippet.additionalData["matchType"]]
 [#assign snippetFilePath = snippet.location.path!""]
-[#if matchType == "PARTIAL" && snippetFilePath?has_content]
+[#if matchType == "PARTIAL" && snippetFilePath?has_content && snippet.additionalData['matchedLinesSnipped']?has_content]
```

Comment:
This behavior that you now need to have matched lines for the snippet report was known for me, unfortunately it was on my TODO list.
My idea was to make the get matched lines option always enabled as is it required anyway to group snippets by source location.
If the performance cost of listing matched lines is problematic, maybe add an option "fetch snippets results": if enabled, the snippets with their matched lines you be listed. Hence, snippets and matched lines are always "enable together".

What do you think ?

Reply:
No objections, although I cannot judge the possible performance penalty. Nevertheless, I think that the change made here is rather independent from your proposed enhancement; it is just an additional safe guard preventing the reporter to crash when there is unexpected input.